From 6add4e4f0040693e7c4242fbae48c9d32007686c Mon Sep 17 00:00:00 2001
From: Mamoru TASAKA <mtasaka@fedoraproject.org>
Date: Fri, 3 Feb 2023 08:11:04 +0900
Subject: [PATCH] lxqt-config-monitor: add more header file inclusion for
 libkscreen 5.26.90 (#915)

With https://github.com/KDE/libkscreen/commit/94f330959b0eda775418aef7faee80ce69144e63 ,
`#include <KScreen/Output>` no longer includes "mode.h" implicitly.
So in lxqt-config-monitor, files using `class KScreen::Mode` should include
`#include <KScreen/Mode>` explicitly.

Related: #903 .
---
 lxqt-config-monitor/kscreenutils.cpp          | 1 +
 lxqt-config-monitor/loadsettings.cpp          | 1 +
 lxqt-config-monitor/monitorpicture.cpp        | 1 +
 lxqt-config-monitor/monitorsettingsdialog.cpp | 1 +
 lxqt-config-monitor/monitorwidget.cpp         | 1 +
 5 files changed, 5 insertions(+)

diff --git a/lxqt-config-monitor/kscreenutils.cpp b/lxqt-config-monitor/kscreenutils.cpp
index 9515e789..be2634d7 100644
--- a/lxqt-config-monitor/kscreenutils.cpp
+++ b/lxqt-config-monitor/kscreenutils.cpp
@@ -2,6 +2,7 @@
 #include "timeoutdialog.h"
 
 #include <KScreen/Output>
+#include <KScreen/Mode>
 #include <KScreen/Config>
 #include <KScreen/GetConfigOperation>
 #include <KScreen/SetConfigOperation>
diff --git a/lxqt-config-monitor/loadsettings.cpp b/lxqt-config-monitor/loadsettings.cpp
index 0c7bd73c..4e9331ba 100644
--- a/lxqt-config-monitor/loadsettings.cpp
+++ b/lxqt-config-monitor/loadsettings.cpp
@@ -23,6 +23,7 @@
 #include "kscreenutils.h"
 #include <KScreen/Output>
 #include <KScreen/Config>
+#include <KScreen/Mode>
 #include <KScreen/GetConfigOperation>
 #include <KScreen/SetConfigOperation>
 #include <LXQt/Settings>
diff --git a/lxqt-config-monitor/monitorpicture.cpp b/lxqt-config-monitor/monitorpicture.cpp
index 0d06ab82..4cb14894 100644
--- a/lxqt-config-monitor/monitorpicture.cpp
+++ b/lxqt-config-monitor/monitorpicture.cpp
@@ -24,6 +24,7 @@
 #include <QDebug>
 #include <QVector2D>
 #include <QRectF>
+#include <KScreen/Mode>
 #include <QScrollBar>
 
 #include "configure.h"
diff --git a/lxqt-config-monitor/monitorsettingsdialog.cpp b/lxqt-config-monitor/monitorsettingsdialog.cpp
index 6172019f..bfd8c1dd 100644
--- a/lxqt-config-monitor/monitorsettingsdialog.cpp
+++ b/lxqt-config-monitor/monitorsettingsdialog.cpp
@@ -28,6 +28,7 @@
 #include "kscreenutils.h"
 
 #include <KScreen/Output>
+#include <KScreen/Mode>
 #include <QJsonObject>
 #include <QJsonArray>
 #include <LXQt/Settings>
diff --git a/lxqt-config-monitor/monitorwidget.cpp b/lxqt-config-monitor/monitorwidget.cpp
index e0fcf0a8..41883c25 100644
--- a/lxqt-config-monitor/monitorwidget.cpp
+++ b/lxqt-config-monitor/monitorwidget.cpp
@@ -22,6 +22,7 @@
 #include <QComboBox>
 #include <QStringBuilder>
 #include <QDialogButtonBox>
+#include <KScreen/Mode>
 #include <KScreen/EDID>
 
 #include <algorithm>
